### ISSUE TITLE
Getting invalid_username_or_password 

### ISSUE THREAD

**Issue by raime910** [IdentityServer/IdentityServer4#2957]
I'm writing an integration test for my microservices authentication and authorization. I am using IdentityServer4 with in-memory stores and my code looks like this; 

	var builder = services
		.AddIdentityServer(options =>
		{
			options.IssuerUri = "https://my.services.is4host";
			options.Events.RaiseErrorEvents = true;
			options.Events.RaiseInformationEvents = true;
			options.Events.RaiseFailureEvents = true;
			options.Events.RaiseSuccessEvents = true;
		});

	var migrationAssembly = typeof(Startup).GetTypeInfo().Assembly.GetName().Name;	
	
	var inMemorySqlite = new SqliteConnection("Data Source=:memory:");
	inMemorySqlite.Open();

	services
		.AddDbContext<SecurityDbContext>(
			options => options.UseSqlite(inMemorySqlite,
				sql => sql.MigrationsAssembly(migrationAssembly)));
				
	// Migrate the database
	var context = services.BuildServiceProvider().GetService<SecurityDbContext>();
	context.Database.Migrate();

	builder
		.AddInMemoryClients(IS4Config.GetClients())
		.AddInMemoryIdentityResources(IS4Config.GetIdentityResources())
		.AddInMemoryApiResources(IS4Config.GetAPIResources())
		.AddTestUsers(new List<TestUser>
		{
				new TestUser {
					SubjectId = Guid.NewGuid().ToString(),
					Claims = new List<Claim> {
						new Claim(JwtClaimTypes.Subject, MyDemoUser.ElevatedUser.UserName),
						new Claim(JwtClaimTypes.Email, MyDemoUser.ElevatedUser.Email),
						new Claim(JwtClaimTypes.Role, MyDemoUser.ElevatedUser.Roles[0]),
						new Claim(JwtClaimTypes.Role, MyDemoUser.ElevatedUser.Roles[1])
					},
					IsActive = true,
					Password = MyDemoUser.ElevatedUser.Password,
					Username = MyDemoUser.ElevatedUser.UserName
				},
				new TestUser {
					SubjectId = Guid.NewGuid().ToString(),
					Claims = new List<Claim> {
						new Claim(JwtClaimTypes.Subject, MyDemoUser.WorkstationUser.UserName),
						new Claim(JwtClaimTypes.Email, MyDemoUser.WorkstationUser.Email),
						new Claim(JwtClaimTypes.Role, MyDemoUser.WorkstationUser.Roles[0])
					},
					IsActive = true,
					Password = MyDemoUser.WorkstationUser.Password,
					Username = MyDemoUser.WorkstationUser.UserName
				}
		});

and my test looks like this;

	var is4Host = new WebApplicationTestFactory<Services.IS4Host.Startup>();
	var client = is4Host.CreateClient();

	var disco = await client.GetDiscoveryDocumentAsync("https://My.services.is4host");
	if (disco.IsError)
	{
		Console.WriteLine(disco.Error);
		return false;
	}

	// request token
	var tokenResponse = await client.RequestPasswordTokenAsync(new PasswordTokenRequest
	{
		Address = disco.TokenEndpoint,
		ClientId = MyClient.ClientId,
		ClientSecret = MyClient.ClientSecret,
		Scope = MyApiResource.Gateway.Name,
		UserName = user.UserName,
		Password = user.Password
	});

	if (tokenResponse.IsError)
	{
		Console.WriteLine(tokenResponse.Error);
		return false;
	}

	Console.WriteLine(tokenResponse.Json);
	Console.WriteLine("\n\n");

	return true;

I am adding records into the in memory stores but I keep getting the response below from IS4. The user.UserName and user.Password that is used for creating the tokenResponse object is the same static object that was added into the TestUser collection when setting up IS4. 

	{
	  "error": "invalid_grant",
	  "error_description": "invalid_username_or_password"
	}

**Comment by brockallen** [comment]
Any update? If not, then, as always, check the logs.

**Comment by raime910** [comment]
I changed my approach to take out the AddTestUsers and do my own seeding into the SecurityDbContext.
And as long as I am using the same connection instance, it works. So I implemented a singleton pattern for the in memory SQLite connection.

I feel like my approach is too "hacky" (for lack of a better term). So I was wondering if this is even the right way to do integration test in IS4 using a custom DbContext?
